internal/k8shandler/collection.go

```diff
@@ -396,27 +397,36 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateCollectorPrometheusRu
 	})
 }
 
-func (clusterRequest *ClusterLoggingRequest) createOrUpdateCollectorConfigMap(collectorType logging.LogCollectionType, collectorConfig string) error {
-	var collectorConfigMap *corev1.ConfigMap = nil
+func (clusterRequest *ClusterLoggingRequest) createOrUpdateCollectorConfigMapSecret(collectorType logging.LogCollectionType, collectorConfig string) error {
```

Comment:
This name should be more generic to not ref "configmap" or "secret".  What is it doing?  its "ReconcilingCollectorConfig".  Actually this is exactly what I referred to where we should consider refactoring this into a collector specific function that does one thing or the other depending upon the collector impl instead of using an if block

Reply:
@jcantrill  This method creates configmap(fluentd) or secret (vector).  I think in Java we can do using `interface` but in golang how do we achieve this without using if block. Can you please share an example ?

Reply:
Golang interface can be used similar to Java to provide alternate implementations of the same methods.

Reply:
thanks @alanconway. Yes Golang interface can work like Java interface. 
@jcantrill  You are right we would need to refactor code to make code extensible and have separate implementations  for  each collector type. This will require some effort to understand in terms of flow to generate collector instance using factory and pass it to these methods. Can we avoid it in this PR as we are about to finish sprint ?  We can do it in separate PR.

Reply:
@ajaygupta978 even if re factoring is not possible, please consider renaming the method

Reply:
Ok done.